third_party/move/tools/move-package/src/source_package/std_lib.rs

```diff
@@ -0,0 +1,98 @@
+// Copyright (c) Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::source_package::parsed_manifest::{Dependency, GitInfo};
+use move_command_line_common::env::MOVE_HOME;
+use move_symbol_pool::symbol::Symbol;
+use std::path::PathBuf;
+
+/// Represents a standard library.
+pub enum StdLib {
+    AptosFramework,
+    AptosStdlib,
+    MoveStdlib,
+}
+
+impl StdLib {
+    /// The well-known git URL for the standard library.
+    const STD_GIT_URL: &'static str = "https://github.com/aptos-labs/aptos-core.git";
+
+    /// Returns the dependency for the standard library with the given version.
+    pub fn dependency(&self, version: &StdVersion) -> Dependency {
+        let local =
+            PathBuf::from(MOVE_HOME.clone()).join(format!("{}_{}", self.as_str(), version.rev()));
+        Dependency {
+            local: local.join(self.sub_dir()),
+            subst: None,
+            version: None,
+            digest: None,
+            git_info: Some(GitInfo {
```

Comment:
@wrwg Just to double check: what is the local caching behavior of the git dependencies? Do we update every time the build command is run, or it has to be triggered by some other events?

Specifically I want to understand what's going to happen when we update the branches in the repo.

Reply:
It depends on whether you set the `skip_fetch_latest_git_deps` command line argument.

Reply:
I think its a bit different, skip will actually skip checks, and otherwise we do update. The code is quite convoluted here. https://github.com/aptos-labs/aptos-core/blob/main/third_party/move/tools/move-package/src/resolution/resolution_graph.rs#L558